cmake/BundledAzureSdkCpp.cmake

```
@@ -16,8 +16,13 @@
 # under the License.
 
 function(use_bundled_libazure SOURCE_DIR BINARY_DIR)
-    set(PATCH_FILE "${SOURCE_DIR}/thirdparty/azure-sdk-cpp-for-cpp/azure-sdk-for-cpp-old-compiler.patch")
-    set(PC "${Patch_EXECUTABLE}" -R -p1 -s -f --dry-run -i "${PATCH_FILE}" || "${Patch_EXECUTABLE}" -p1 -i "${PATCH_FILE}")
+    set(PATCH_FILE1 "${SOURCE_DIR}/thirdparty/azure-sdk-cpp-for-cpp/azure-sdk-for-cpp-old-compiler.patch")
+    set(PATCH_FILE2 "${SOURCE_DIR}/thirdparty/azure-sdk-cpp-for-cpp/fix-illegal-qualified-name-in-member.patch")
+    set(PC bash -c "set -x && (\"${Patch_EXECUTABLE}\" -p1 -N -i \"${PATCH_FILE1}\" &&\
+            \"${Patch_EXECUTABLE}\" -p1 -N -i \"${PATCH_FILE2}\") ||\
+            (\"${Patch_EXECUTABLE}\" -p1 -R --dry-run -i \"${PATCH_FILE1}\" &&\
+            \"${Patch_EXECUTABLE}\" -p1 -R --dry-run -i \"${PATCH_FILE2}\")")
```

Isn't this backwards?  We want to "dry-run" the reverse patch first, and only run the real patch if the dry-run failed:

```suggestion
    set(PC bash -c "set -x &&\
            (\"${Patch_EXECUTABLE}\" -p1 -R --dry-run -i \"${PATCH_FILE1}\" || \"${Patch_EXECUTABLE}\" -p1 -N -i \"${PATCH_FILE1}\") &&\
            (\"${Patch_EXECUTABLE}\" -p1 -R --dry-run -i \"${PATCH_FILE2}\")" || \"${Patch_EXECUTABLE}\" -p1 -N -i \"${PATCH_FILE2}\") ")
```

---

I copied this from OSSPUUID. The end result is the same, but if you think this is better for some reason, I can apply the change.

---

My version ("for each patch: check if the patch has been already applied, and if not, then apply it") makes more sense to me, but this is subjective.  Having read it five more times, I can see now that your (or OSSPUUID's) version of "try to apply both patches, and if at least one of them fails, then check if both patches have been already applied at this point, and if not, then give an error" is logically the same, so I'm OK with either.

---

I agree that your version might be easier to follow, so changed it. Thanks.